Tribler/Test/API/test_bootstrap.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import
+
+import logging
+import os
+from binascii import hexlify
+
+from twisted.internet.defer import Deferred
+from twisted.internet.defer import inlineCallbacks
+
+from Tribler.Core.Utilities.bootstrap_util import create_dummy_tdef
+from Tribler.Core.simpledefs import DLSTATUS_SEEDING, dlstatus_strings
+from Tribler.Test.test_as_server import TESTS_DATA_DIR
+from Tribler.Test.test_as_server import TestAsServer
+from Tribler.Test.tools import trial_timeout
+
+
+class MockPayoutManager(object):
+
+    def __init__(self):
+        self.peers = {}
+
+    def update_peer(self, mid, balance):
+        self.peers[mid] = balance
+
+    def do_payout(self):
+        return self.peers
+
+
+class TestBootSeed(TestAsServer):
+
+    @inlineCallbacks
+    def tearDown(self):
+        os.remove(self.sourcefn)
+        yield super(TestBootSeed, self).tearDown()
+
+    @inlineCallbacks
+    def setUp(self):
+        yield super(TestBootSeed, self).setUp()
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.test_deferred = Deferred()
+        self.sourcefn = os.path.join(TESTS_DATA_DIR, 'bootstrap.block')
+        self.tdef = create_dummy_tdef(self.sourcefn, 25)
+        self._logger.debug("Creating file with infohash %s", hexlify(self.tdef.infohash))
+        self.count = 0
+        self.payout_manager = MockPayoutManager()
+
+    def setUpPreSession(self):
+        super(TestBootSeed, self).setUpPreSession()
+        self.config.set_libtorrent_enabled(True)
+        self.config.set_libtorrent_max_download_rate(1)
+
+    def start_download(self, dscfg):
+        download = self.session.start_download_from_tdef(self.tdef, dscfg)
+        download.set_state_callback(self.downloader_state_callback)
+        download.add_peer(("127.0.0.1", self.seeder_session.config.get_libtorrent_port()))
+
+    @trial_timeout(20)
+    def test_seeding(self):
+        """
+        Test whether a torrent is correctly seeded
```

Comment:
This is not only testing for correct seeding behaviour right? Since the goal of this test is to test the payout mechanism, I would rename this test to `test_bootstrap` and change the docstring.